client/components/changed-badge.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+class ChangedBadge extends React.Component {
+  render() {
+    if (this.props.fields.some( k=> this.props.latest.some( l=> l.includes(k)))) {
+      return <span className="badge changed">changed</span>;
+    }
+    if (this.props.fields.some( k=> this.props.granted.some( l=> l.includes(k)))) {
+      return <span className="badge">amended</span>;
+    }
+    return null;
+  }
+}
+
+export default ChangedBadge;
```

Comment:
This file doesn't need a fully fledged React.Component - it doesn't have any lifecycle hooks etc so a plain old function would be better in this case.